src/main/scala/com/cloudant/clouseau/ClouseauQueryParser.scala

```diff
@@ -62,7 +62,7 @@ class ClouseauQueryParser(version: Version,
                              startInclusive: Boolean,
                              endInclusive: Boolean): Query = {
     if (isNumber(lower) && isNumber(upper)) {
-      NumericRangeQuery.newDoubleRange(field, 8, lower.toDouble,
+      NumericRangeQuery.newDoubleRange(field, 16, lower.toDouble,
```

Comment:
This triggered some investigation on my part, coupled with the other comments about precision step default change.

Seems we currently index with precision step of 4 (default for 4.6.1) but query with precision step 8. That's probably a bug, albeit one we've had for a long time.

I don't think we can change this when we upgrade as it might cause search results to change.

Instead, we need to keep the above as `8` and change how we _index_ doubles to continue to do so with precision step of 4 not the 'new' default of 16.

Reply:
See https://github.com/cloudant-labs/clouseau/commit/742670bc9395ac458edbcbc1fdc8639610634dcb for my attempt at doing so.

Reply:
https://github.com/cloudant-labs/clouseau/commit/3519e6803f97ad8596b26bf72bb40472acef3b2a rather

Reply:
no, it means that search results could change. some double values will be quantised to different values before and after this change. Existing indexes will have a mixture of the old and new precision step. Simply put, we can change precision step (and other things) but it requires an index rebuild to get the right answers.

Reply:
I looked into it further, I think we would benefit from indexing with precision step `8`, given that we have (mistakenly) been using that value in all numeric range queries. Creating extra terms with precision step `4` is wasteful as we never use them for searching.

```
tokens created for precision step 4;
0 4653142547559993678
4 4653142547559993664
8 4653142547559993600
12 4653142547559993344
16 4653142547559940096
20 4653142547559874560
24 4653142547550437376
28 4653142547349110784
32 4653142546006933504
36 4653142485877391360
40 4653142004841054208
44 4653133208748032000
48 4653062840003854336
52 4652218415073722368
56 4611686018427387904
60 4611686018427387904

tokens created for precision step 8;
0 4653142547559993678
8 4653142547559993600
16 4653142547559940096
24 4653142547550437376
32 4653142546006933504
40 4653142004841054208
48 4653062840003854336
56 4611686018427387904
```

Reply:
also quoting from 5609;

```To be simple, i think a change we make has to be a multiple of 4: the current default.

This way we have easy backwards compat with existing indexes for users that just go with the default (since NumericRangeQuery has ctors that use the default precisionStep, and multiples of the index-time work at query-time).```

Reply:
Sorry, I do not see how does all this translates to the current situation.  Should we change the `Double` field type wrapper to set the numeric precision step to `8` instead of `4`?

Reply:
Sorry, there are several issues here.

Clouseau has apparently always been wrong on this. We are indexing numeric values with precision step of 4 but only ever querying with precision step of 8. Half of those terms are never queried, there's no possible path to them, but we took the time to index them and store them.

Later version of Lucene have changed the default (from 4 to 16), which triggered this conversation.

I don't think we should change to 16 as it would change the results of current queries (I think, but this should be challenged, perhaps I misunderstand the implications).

So, the least we must do is prevent the default from changing to 16, and that requires the FieldType changes I show in my commit above.

The only question remaining is whether we keep exactly what we have today (4 for indexing, 8 for querying), or whether we address that mistake and switch to 8 for indexing and querying.

If I'm wrong about the implications of 16 then we could of course do nothing, but someone other than me will need to do that, or it's just an echo chamber.

Specifically, I worry that a document might no longer match a given query if we go from 4 to 16 because the numeric term that would match in the former case simply doesn't exist in the latter case.

(as a primer, the way old versions of Lucene index numbers is to convert them into a series of specially formatted _strings_ which are then indexed as terms. Newer versions of Lucene use a better system called `points` and the notion of precision step no longer applies).

Reply:
Noting there's a good chance that precision step is "just" about performance. the actual numeric value is indexed in all cases, but not all of the speed-up / lower precision terms are. This is all ancient history and poorly remembered (by me).

Reply:
Thank you for the explanation.

This uplift would require a validation to eliminate the generic worries around the potential index format changes anyway.  Once we had the related end-to-end tests implemented, we can also experiment with the various configurations on precision step.